### PR TITLE
[pkg/ottl]: Add MD5 function

### PR DESCRIPTION
**Pull request by edmocosta**
**Description:** <Describe what has changed.>

Introduced the `MD5` converter function which returns the MD5 hash/digest of the given `value` (path expression to a string telemetry field or a literal string). 

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33792

**Testing:** <Describe what testing was performed and which tests were added.>
- Unit tests
- E2E tests

**Documentation:** 
A new README entry was added for the `MD5` function.

### CODE REVIEW

File: pkg/ottl/ottlfuncs/README.md [hunk 1]
```diff
@@ -877,6 +878,26 @@ Examples:
...
+
+If an error occurs during hashing it will be returned.
+
+Examples:
+
+- `MD5(attributes["device.name"])`
+
+- `MD5("name")`
+
+**Note:** According to the National Institute of Standards and Technology (NIST), MD5 is no longer a recommended hash function. It should be avoided except when required for compatibility. New uses should prefer FNV whenever possible.
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/README.md:899]:
```suggestion
**Note:** According to the National Institute of Standards and Technology (NIST), MD5 is no longer a recommended hash function. It should be avoided except when required for compatibility. New uses should prefer a SHA-2 family function (e.g. SHA-256, SHA-512) whenever possible.
```
I'm not sure exactly what to recommend, but FNV is a non-cryptographic hash function, so we should recommend a different cryptographic hash function instead. OTTL at least has SHA-256, and there is an open PR for SHA-512.

File: pkg/ottl/ottlfuncs/README.md [hunk 2]
```diff
@@ -877,6 +878,26 @@ Examples:
...
 
+### MD5
+
+`MD5(value)`
+
+The `MD5` Converter converts the `value` to a md5 hash/digest.
+
+The returned type is string.
+
+`value` is either a path expression to a string telemetry field or a literal string. If `value` is another type an error is returned.
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/README.md:889]: Do we have to limit the input to strings? Could we include numbers and byte slices as well for starters?

Comment (evan-bradley) [reply pkg/ottl/ottlfuncs/README.md:889]: I took a second look and all our other hash functions have the same restriction, let's just leave this for now.

File: pkg/ottl/ottlfuncs/README.md (hunk 1 above)
Comment (evan-bradley) [reply pkg/ottl/ottlfuncs/README.md:899]: I see now that this was just copied from our existing hash functions, sorry for the trouble there. I've updated our other hash function notes as well: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34359

**Review by evan-bradley — APPROVED** [review]
We'll want to update the hash function alternative recommendation, but otherwise this looks good to me. Thanks @edmocosta.

**Review by atoulme — APPROVED** [review]

**Comment by edmocosta** [comment]
> We'll want to update the hash function alternative recommendation, but otherwise this looks good to me. Thanks @edmocosta.

Thanks for reviewing folks! I've applied @evan-bradley suggestion to update the alternative recommended hash function.

**Comment by evan-bradley** [comment]
@TylerHelmuth please take a look when you are able.

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by evan-bradley** [comment]
@TylerHelmuth Please take a look if you have time today, it would be nice to get this in for the next release.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**